tasks/static_quality_gates/gates.py

```diff
@@ -0,0 +1,866 @@
+from tasks.libs.package.size import InfraError, directory_size, extract_package, file_size
+
+# Architecture definitions are different depending on OS
+ARCH_MAPPING = {
+    "amd64": "x86_64",
+    "arm64": "aarch64",
+    "armhf": "armv7hl",
+}
+
+PACKAGE_OS_MAPPING = {
+    "deb": "debian",
+    "rpm": "centos",
+    "suse": "suse",
+    "heroku": "debian",
+    "msi": "windows",
+}
+
+
+def byte_to_string(size: int, unit_power: int = None, with_unit: bool = True) -> str:
+    """
+    Convert a byte size to a string with unit suffix.
+    param: size: the size along with the unit suffix
+    param: unit_power: the power of the unit to use
+    param: with_unit: whether to include the unit suffix in the returned string
+    return: size as a string
+    """
+    if not size:
+        return f"0{' B' if with_unit else ''}"
+    sign = ""
+    if size < 0:
+        size *= -1
+        sign = "-"
+    size_name = ("B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
+    if unit_power is None:
+        unit_power = int(math.log(size, 1024))
+    p = math.pow(1024, unit_power)
+    s = round(size / p, 2)
+    # If s is not exactly 0 but rounded like (0.0 or -0.0)
+    # Goal is to output +0 / -0 for very small changes and 0 for no changes at all
+    if id(s) != id(0) and s == 0:
+        s = 0
+    return f"{sign}{s}{' ' + size_name[unit_power] if with_unit else ''}"
+
+
+def string_to_latex_color(text: str) -> str:
+    """
+    Convert a string to a latex color.
+    param: text: the text to convert
+    return: the text as a latex color
+    """
+    # Github latex colors are currently broken, we are disabling this function's color temporarily for now
+    return r"$${" + text + "}$$"
+
+
+def string_to_byte(size: str) -> int:
+    """
+    Convert a string to a byte size.
+    param: size: the size as a string
+    return: the size in bytes
+    """
+    if not size:
+        return 0
+    size_name = ("KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
+    value = None
+    power = 0
+    for k, unit in enumerate(size_name):
+        if unit in size:
+            value = float(size.replace(unit, ""))
+            power = k + 1  # We start with KiB = 1024^1
+            break
+    if value:
+        return int(value * math.pow(1024, power))
+    elif "B" in size:
+        return int(float(size.replace("B", "")))
+    else:
+        return int(size)
+
+
+def read_byte_input(byte_input: str | int) -> int:
+    """
+    Read a byte input and return the size in bytes.
+    param: byte_input: the byte input
+    return: the size in bytes
+    """
+    if isinstance(byte_input, str):
+        return string_to_byte(byte_input)
+    else:
+        return byte_input
+
+
+class StaticQualityGateFailed(Exception):
+    """
+    Exception raised when a static quality gate fails
+    """
+
+    def __init__(self, message: str):
+        self.message = color_message(message, "red")
+        super().__init__(self.message)
```

Comment:
The `message` attribute was intentionally removed from Python 3. To retrieve the message from an exception `e`, the need is now covered with `str(e)`.
If one still wants to achieve `e.message` despite the above mentioned design choice, one may add a property instead, for instance:
```suggestion
        super().__init__(color_message(message, "red"))

    @property
    def message(self):
        return self.args[0]
```